pkg/runtest/run.go

```diff
@@ -204,6 +204,12 @@ func progFileList(dir, filter string) ([]string, error) {
 func (ctx *Context) generateFile(sandboxes []string, cover []bool, filename string) error {
 	p, requires, results, err := parseProg(ctx.Target, ctx.Dir, filename, nil)
 	if err != nil {
+		for _, sandbox := range sandboxes {
+			ctx.createTest(&runRequest{
+				name:   fmt.Sprintf("%v %v", filename, sandbox),
+				broken: fmt.Sprintf("%v", (err)),
```

Comment:
Broken program don't lead to test failure, and there is a number of these (so any new ones will be lost among existing broken tests).
If I add a broken program that does not parse, the test will say OK.
Is it a reasonable behavior?

Reply:
I think it is. There aren't any broken programs right now, and the tests also report the broken count separately, so it can be noticed.

If you think a broken program should indicate failure, we should check for `fail != 0 || broken != 0` in `Run()`.

Reply:
There should be some broken now for other OSes.
It's intentional to not fail on known broken programs, otherwise it's not possible to have known broken cases and pass tests. 
This case is different, it not a know broken program (it does not make sense to have non-parsing programs in tests).

If you want to handle broken programs w/o stopping all testing, I think you need to a new notion of programs that failing after the parsing stage (don't need to try to execute them), then the loop will (1) print "FAIL" instead of "BROKEN" for them, and (2) increment fail rather than broken.

Reply:
Ok, that makes sense, let me add another member to runRequest to handle this case.